glean-core/python/glean/glean.py

```diff
@@ -220,12 +233,25 @@ def _initialize_core_metrics(cls):
         """
         Set a few metrics that will be sent as part of every ping.
         """
-        from . import _builtins
+        from ._builtins import metrics
+
+        metrics.glean.baseline.locale.set(util.get_locale_tag())
+        metrics.glean.internal.metrics.os.set(platform.system())
+        metrics.glean.internal.metrics.os_version.set(platform.release())
+        metrics.glean.internal.metrics.architecture.set(platform.machine())
+
+        # device_model and device_manufacturer aren't easily obtainable in this
```

Comment:
Are these required in the Glean schema?

Reply:
Indeed they are.  So I guess we'll have to put something here...

Reply:
I'm putting "unknown" in here for now, so it passes the schema.  I don't think there's really a cross-platform way to get the manufacturer/model of the desktop computer.  We can get CPU architecture and such, and OS version things (which are already reported in other fields).

We could adjust the schema to not require these...

Reply:
> 
> 
> I'm putting "unknown" in here for now, so it passes the schema. I don't think there's really a cross-platform way to get the manufacturer/model of the desktop computer. We can get CPU architecture and such, and OS version things (which are already reported in other fields).

Sounds good to me for the MVP!

> We could adjust the schema to not require these...

Yeah, I suspect this might be a schema-v2 thing.. Any chance you could file a bug, tag it with schema-v2, add a comment to this part of the code with the bug number?

Reply:
I did a little poking around after I wrote this yesterday.  It turns out the information exists in the hardware BIOS.  There just doesn't seem to be a cross-platform way to get at it, so we would probably have to build it.  I agree we don't need any of this for MVP.

But I reworked the bug to be [add device info to Python bindings](https://bugzilla.mozilla.org/show_bug.cgi?id=1595751), rather than removing them from the schema.